File: MatlabOctaveCodeSnippets/snip_signals_noisy_wavread.m
fp=fopen('myvoice.wav','rb'); %open for reading in binary
x=fread(fp,Inf,'int16'); %read all samples as signed 16-bits
fclose(fp);  %close the file
x(1:22)=[]; %eliminate the 44-bytes header
[x2,Fs,b]=readwav('myvoice.wav','r');
x2=double(x2); %convert integer to double for easier manipulation
max(abs(x-x2)) %result is 0, indicating they are identical

File: PythonCodeSnippets/snip_signals_noisy_wavread.py
import numpy as np
from scipy.io import wavfile

# read the wav file
with open('myvoice.wav', 'rb') as fp:
    x = np.fromfile(fp, dtype=np.int16)

fp.close() # close the file
x = x[39:]  # remove the header

# read the wav file again for comparison
Fs, x2 = wavfile.read('myvoice.wav')
x2 = x2.astype(float)  # convert integer to double for easier manipulation

# check if they are identical
print(np.max(np.abs(x - x2)))  # result is 0, indicating they are identical